.github/assets/check_rv32im.sh

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env bash
+set +e  # Disable immediate exit on error
+
+# Array of crates to compile
+crates=($(cargo metadata --format-version=1 --no-deps | jq -r '.packages[].name' | grep '^reth' | sort))
+
+# Array of crates to exclude
+# Used with the `contains` function.
+# shellcheck disable=SC2034
+exclude_crates=(
```

Comment:
can we invert this please, 

checking crates for inclusion is pretty annoying, because what we need to do here now is check that a selection of crates compiles for this target, then we don't need the cargo metadata call